app/models/spotlight/resources/oaipmh_mods_parser.rb

```diff
@@ -55,43 +55,52 @@ def parse_mods_record
 
     def uppercase_unique_id
       # check if the unique ID has a lowercase letter in it
-      if (solr_hash["unique-id_tesim"] =~ /[a-z]/).present?
+      if (solr_hash['unique-id_tesim'] =~ /[a-z]/).present?
         # make all URNs uppercase
-        solr_hash["unique-id_tesim"] = solr_hash["unique-id_tesim"].upcase!
+        solr_hash['unique-id_tesim'] = solr_hash['unique-id_tesim'].upcase!
       end
     end
 
-    def search_id
+    def search_id(exhibit_id)
       # strip out decimal and store in hashes
-      solr_hash['search-id_tesim'] = sidecar_data['search-id_tesim'] = solr_hash['unique-id_tesim'].gsub('.', '')
+      id_arr = []
+      id_arr << solr_hash['unique-id_tesim'].gsub('.', '') # stripped_decimals
+      id_arr << "#{exhibit_id}-#{solr_hash['unique-id_tesim'].gsub('.', '')}" # stripped_decimals w/ exhibit id
+      id_arr << solr_hash['unique-id_tesim'].gsub('.', '').gsub(':', '') # all_punc_stripped
+      id_arr << "#{exhibit_id}-#{solr_hash['unique-id_tesim'].gsub('.', '').gsub(':', '')}" # all_punc_stripped w/ exhibit id
+      id_arr << urn = fetch_ids_uri(@item_solr['thumbnail_url_ssm']).split('/').last.split('?').first # urn
+      id_arr << urn.gsub('.', '').gsub(':', '') # urn with punc stripped
+
```

Comment:
Reading note: we're ensuring that searches via multiple nuanced differences work.